### PR TITLE
[avro] Support nullable multi-type unions as Variant (#799)

### LINKED ISSUE #799: [avro] Nullable unions
### Description 

nullable unions (see: src/testFixtures/avro/schemas/incompatible/union_null_string_int.json)

### PR DESCRIPTION
**Pull request by 29shivam**
## Summary

Fixes #799 (part of #726).

Confluent's Avro converter turns a multi-branch union like `[null, string, int]` into an **optional** Connect union struct (`io.confluent.connect.avro.Union`) with one optional field per non-null branch. The connector already handles this end to end:
- `Column.connectTypeToClickHouseType` maps the union struct → `Variant(String, Int32)`.
- The RowBinary Variant write path picks the non-null branch, resolves its global discriminator (case-insensitively matching the Connect type to the ClickHouse variant type), and writes discriminator + value — or the ClickHouse null discriminator (`255`) when no branch is set.

Per @chernser's guidance, `Variant` is the right target here since the branches are simple types (`String`, `Int32`).

Changes:
- **Add `AvroNullableUnionVariantTest`** — drives the string branch, the int branch, and the all-null case through the convert (`StructToJsonMap`) + Variant write path, asserting the discriminator + value bytes. Runs without a live ClickHouse.
- **Promote `union_null_string_int.json`** from `incompatible/` to `compatible/` so the Avro integration test exercises it end to end.
- CHANGELOG entry under `1.4.1 (unreleased)`.

**Verification:** feature test passes locally; the full end-to-end path runs via the integration test in CI.

## Checklist
Delete items not relevant to your PR:
- [x] Unit and integration tests covering the common scenarios were added
- [x] A human-readable description of the changes was provided to include in CHANGELOG

### CODE REVIEW

**Comment by 29shivam** [comment]
@chernser ready for review — nullable multi-type unions (#799). Per your note, `Variant` fits here since the branches are simple types (`String`, `Int32`). Feature test drives the string/int/all-null cases through the Variant write path; fixture promoted to `compatible`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+# 1.4.1, (unreleased)
```
Comment (chernser) [inline CHANGELOG.md]: please update to latest one

**Review by chernser — CHANGES_REQUESTED** [review]
Is there code changes? I see only test. 
tests should cover more unions - because not all might fit Variant.

**Comment by 29shivam** [comment]
Thanks @chernser!

- **Code vs test:** you're right — there's no new *production* code here. The union→Variant path already exists (`connectTypeToClickHouseType`/`resolveUnionType` map the Confluent union struct to `Variant(...)`, and the RowBinary Variant write path resolves the per-value discriminator), so this fixture was already handled correctly. The PR's value is promoting it from `incompatible/` → `compatible/` so the integration test exercises it, plus locking in coverage. If you'd rather not have a test-only PR, I'm happy to fold this into another PR or close it — your call.

- **More union coverage:** expanded the test — it now covers `Variant(String, Int32)` (string / int / all-null), `Variant(String, Int32, Bool)` (bool branch + discriminator), and a suspicious same-group numeric union `[int, long]` that falls back to `Nullable(String)` rather than a Variant — so 'not everything fits a Variant' is covered.

- **CHANGELOG:** rebased onto latest `main`; the entry now sits under `1.5.0 (not published)`.

**Comment by 29shivam** [comment]
The one red CI cell (`one_shard_three_replicas 26.3 V1`) is a runner/infra failure, not this change — it died at container startup with `Can't get Docker image: testcontainers/ryuk:0.14.0` and the subsequent `docker compose up` failing, so every container-based test errored at init (all `initializationError`, no assertions). The other 32 cells are green, including the cluster/standalone cells that run the Avro integration test with the promoted `union_null_string_int` fixture — so the fixture passes end to end. Could you re-run that one job when you get a chance? 🙏

**Review by chernser — APPROVED** [review]

**Comment by 29shivam** [comment]
Thank you @chernser.
